server.js

```diff
@@ -42,6 +42,12 @@ if (fs.existsSync('.env')) {
   process.exit(1)
 }
 
+if (config.private.redis_url != null) {
+  console.warning(
+    'RedisTokenPersistence is deprecated for token pooling and will be removed in a future release. Migrate to SqlTokenPersistence'
+  )
+}
+
```

Comment:
I decided that it is useful to keep both backends around for some length of time because it allows us to switch between them just by changing an env var - no code changes needed. Doing this means we can:

- deploy this code with `REDIS_URL` set (so we're still using the redis backend)
- migrate our data to postgres with `scripts/migrate-token-pool.js`
- set `POSTGRES_URL` to switch to the postgres backend
- if it all goes wrong, we can easily just unset `POSTGRES_URL` to go back to redis
- unset `REDIS_URL` once we're happy everything is fine
- delete the redis code at a later date

This both makes the migration super low-risk/easy to back out of and means we can provide a migration path for self-hosting users with basically no effort.

Reply:
Might be worth extending the warning message to include a link to https://github.com/badges/shields/issues/8921

Reply:
Yes. I'll call this out at the top of the changelog for the next server release and write it with a bit more of a focus on self-hosting users than the discussion in https://github.com/badges/shields/issues/8921 (which is quite shields.io-specific). I'm going to set myself a reminder to ping me so I remember to do this when I do the changelog.

@SnoozeThis wait until Feb 28 2023